### PR TITLE
fix: use max debt limit

### PR DESCRIPTION
**Pull request by samsiegart**
- Use full debt limit now that we have: https://github.com/Agoric/agoric-sdk/pull/6984
- Fix rounding error for "max" button in vault creation and added test.

### CODE REVIEW

File: test/utils/vaultMath.test.ts [hunk 1]
```diff
@@ -232,30 +232,30 @@ describe('maxCollateralForNewVault', () => {
   it('should prevent the user from exceeding the mint limit', () => {
```
Comment (samsiegart) [inline test/utils/vaultMath.test.ts:232]: Confirmed this test now fails without change. We also have https://github.com/Agoric/dapp-inter/blob/6aa6db5ede8a9586d46fcd295dc50c78d1372ea6/test/utils/vaultMath.test.ts#L19 to test the round-up case.

File: src/utils/vaultMath.ts [hunk 2]
```diff
@@ -49,7 +49,13 @@ export const computeToLock = (
   toReceive: NatValue,
   defaultCollateralization: Ratio,
   loanFee: Ratio,
+  // Round up by default to preserve offer safety. But, if using this to set
+  // max input, round down to avoid hitting debt limit.
+  roundDown: boolean = false,
```
Comment (turadg) [inline src/utils/vaultMath.ts:54]: technically it's not rounding. Let's be more explicit about ceil vs floor. https://github.com/Agoric/agoric-sdk/blob/bcf33feb2ff38d08714c77ef8e644409a4d894fa/packages/zoe/src/contractSupport/safeMath.js#L21-L27

Floor is the default way JS does bigint division (truncating non-integer) so it should be the default. 

```suggestion
  remainderHandling: 'floor' | 'ceil' = 'floor',
```

File: src/utils/vaultMath.ts [hunk 3]
```diff
@@ -159,6 +165,7 @@ export const maxCollateralForNewVault = (
     istAvailableBeforeLoanFee.value,
     desiredCollateralization,
     loanFee,
+    true,
```
Comment (turadg) [inline src/utils/vaultMath.ts:168]: by using the string literal as the option it's readable here what it's doing.
```suggestion
    'ceil',
```

File: test/utils/vaultMath.test.ts (hunk 1 above)
Comment (turadg) [reply test/utils/vaultMath.test.ts:232]: thanks for checking and noting

File: src/utils/vaultMath.ts (hunk 2 above)
Comment (samsiegart) [reply src/utils/vaultMath.ts]: Done

File: src/utils/vaultMath.ts (hunk 3 above)
Comment (samsiegart) [reply src/utils/vaultMath.ts]: Done

**Review by turadg — APPROVED** [review]

**Pull request merged**